jellyfin-api/src/main/kotlin/org/jellyfin/sdk/api/client/extensions/MediaInfoApiExtensions.kt

```diff
@@ -0,0 +1,61 @@
+package org.jellyfin.sdk.api.client.extensions
+
+import org.jellyfin.sdk.api.operations.MediaInfoApi
+import org.jellyfin.sdk.model.BitrateMeasurement
+
+/**
+ * Create a [BitrateMeasurement] for a response of [bytes]. The returned value contains the input, duration and the
+ * bitrate.
+ *
+ * @throws AssertionError If the server response is invalid.
+ */
+@Suppress("MagicNumber")
+public suspend fun MediaInfoApi.measureBitrate(bytes: Int): BitrateMeasurement {
+	val now = System.currentTimeMillis()
+
+	// Request the bits
+	getBitrateTestBytes(size = bytes).apply {
+		// Validate response status
+		assert(status == 200) { "Returned status $status is not OK" }
+
+		// Download full content
+		content.awaitContent()
+	}
+
+	// Calculate duration
+	val duration = System.currentTimeMillis() - now
+
+	// Return results
+	return BitrateMeasurement(
+		bytes = bytes,
+		duration = duration,
+		bitrate = bytes / duration * 8000 // bytes -> bits
+	)
+}
+
+/**
+ * Detect the bitrate of the current connection by calling [measureBitrate] until a measurement of at least 1 second or
+ * with the maximum byte size is created.
+ *
+ * @param minBitrate The resulting [BitrateMeasurement.bitrate] field is at least [minBitrate]. Defaults to 1Mb.
```

Comment:
I don't understand what this is for, what would be a use-case for it?

Reply:
- Prevent too low values in case of weird network conditions
- Set a minimal required bound for the result
- Allow the application to change the minimal value

The default value (1 Mb) is the bitrate similar to 480p h.264 video.

Reply:
Why should the SDK handle that, however? It even violates the bitrate = bytes/duration invariant for the measurement object..
I think it'd make more sense to move that decision to the client, and return the bitrate exactly like it was measured, however good or bad it was (and without caring about the quality of the measurement).
You could just add a comment to the KDoc suggesting using a `max(result, minimum)`.